cg/store/api/status.py

```diff
@@ -620,35 +636,42 @@ def analyses_to_deliver(self, pipeline: Pipeline = None) -> Query:
             .order_by(models.Analysis.uploaded_at.desc())
         )
 
+    def get_report_records_by_pipeline(self, pipeline: Pipeline = None) -> Query:
+        """Fetches the delivery report related records associated to the provided or supported pipelines"""
+
+        records = self.Analysis.query.join(models.Analysis.family)
+        records = (
+            records.filter(models.Analysis.pipeline == str(pipeline))
+            if pipeline
+            else records.filter(models.Analysis.pipeline.in_(REPORT_SUPPORTED_PIPELINES))
+        )
+
+        return records
+
     def analyses_to_delivery_report(self, pipeline: Pipeline = None) -> Query:
-        """Fetch analyses that needs the delivery report to be regenerated."""
+        """Fetches analyses that need a delivery report to be regenerated"""
 
-        analyses_query = self.latest_analyses()
+        records = self.get_report_records_by_pipeline(pipeline)
+        records = records.filter(
+            models.Analysis.delivery_report_created_at.is_(None),
```

Comment:
Same here. It would be great with unit tested, reusable functions that explain what is being done instead of having to read the logic all the time.